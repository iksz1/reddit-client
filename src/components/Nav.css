.nice-menu {
  list-style: none;
  display: flex;
  padding: 2px;
  margin: 0;
  /* background: #dadada; */
  /* border-radius: 0.3125em; */
  cursor: default;
  /* position: absolute;
  font-size: 1.2rem;
  font-weight: 500;
  bottom: calc(-2.5em / 2 - 10px); */
  /* transform-origin: 0 50%; */
  /* transform: scale(0.7); */
}
.nice-menu li {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0.16em;
  width: 2.5em;
  height: 2.5em;
  /* border: 0.3125em solid #767676; */
  border: 0.3125em solid #e2e2e2;
  border-radius: 0.3125em;
  background: #767676;
  color: #fff;
  /* box-shadow: 1px 1px 1px #767676; */
}
.nice-menu li:hover {
  color: #ecb8b8;
}
.nice-menu li.active {
  /* color: #9ebfdf; */
  color: #ecb8b8;
}
