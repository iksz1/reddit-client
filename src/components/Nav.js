import React from "react";
import PropTypes from "prop-types";
// import { connect } from "react-redux";
import "./Nav.css";

export const Nav = ({ items, active, handleClick }) => (
  <ul className="nice-menu">
    {items.map(item => (
      <li
        key={item.name}
        className={`nice-item ${item.name === active ? "active" : ""}`}
        onClick={() => handleClick(item.name)}
      >
        {item.text}
      </li>
    ))}
  </ul>
);

Nav.propTypes = {
  items: PropTypes.array.isRequired,
  handleClick: PropTypes.func.isRequired,
  active: PropTypes.string
};
