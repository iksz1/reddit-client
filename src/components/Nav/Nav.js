import React from "react";
import PropTypes from "prop-types";
import "./Nav.css";

export const Nav = ({ items, active, handleClick }) => (
  <nav>
    <ul className="nice-menu">
      {items.map(item => (
        <li
          key={item.name}
          className={item.name === active ? "active" : ""}
          onClick={() => handleClick(item.name)}
        >
          {item.text}
        </li>
      ))}
    </ul>
  </nav>
);

Nav.propTypes = {
  items: PropTypes.array.isRequired,
  handleClick: PropTypes.func.isRequired,
  active: PropTypes.string
};
