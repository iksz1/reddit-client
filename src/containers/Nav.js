import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import "./Nav.css";

const Nav = ({ items, active, history }) => (
  <ul className="nice-menu">
    {items.map(item => (
      <li
        key={item.name}
        className={`nice-item ${item.name === active ? "active" : ""}`}
        onClick={() => history.push(`/r/${item.name}`)} //prevent repeating
      >
        {item.text}
      </li>
    ))}
  </ul>
);

Nav.propTypes = {
  items: PropTypes.array.isRequired,
  history: PropTypes.object.isRequired,
  active: PropTypes.string
};

const mapState = state => ({
  items: state.subs.items,
  active: state.nav.active
});

export default connect(mapState)(Nav);
