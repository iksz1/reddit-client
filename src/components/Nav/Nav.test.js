import React from "react";
import renderer from "react-test-renderer";
import { Nav } from "./Nav";

const items = [
  { name: "javascript", text: "J" },
  { name: "reactjs", text: "R" },
  { name: "webdev", text: "W" }
];

it("renders correctly", () => {
  const tree = renderer
    .create(<Nav items={items} active={items[0].name} handleClick={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
