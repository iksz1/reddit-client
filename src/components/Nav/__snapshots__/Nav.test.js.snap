// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <ul
    className="nice-menu"
    role="menu"
  >
    <li
      aria-label="javascript"
      className="active"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      J
    </li>
    <li
      aria-label="reactjs"
      className=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      R
    </li>
    <li
      aria-label="webdev"
      className=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      W
    </li>
  </ul>
</nav>
`;
